pennylane/gradients/spsa_gradient.py

```diff
@@ -559,14 +588,33 @@ def _spsa_grad_legacy(
         # Skip the unshifted tape
         shifts = shifts[1:]
 
+    if sampler_seed is not None:
+        if sampler_rng is None:
+            warnings.warn(
+                (
+                    "The sampler_seed argument is deprecated. Use the sampler_rng argument "
+                    "instead to pass a random number generator or a seed."
+                ),
+                UserWarning,
+            )
+            sampler_rng = sampler_seed
+        else:
+            raise ValueError(
+                (
+                    "Both arguments sampler_rng and sampler_seed were specified. "
+                    "Only specify one. Note that the sampler_seed argument is deprecated."
+                )
+            )
+
+    sampler_rng = np.random.default_rng(sampler_rng)
```

Comment:
Makes me think whether we want a `process_sampler_args` method to be called in both versions of `_spsa_grad` :)

Reply:
I think we are getting read of the legacy return type system in the next release cycle, no? Otherwise this would definitely make sense!

Reply:
Yep, true. Still could be nice to factor out. Not crucial though, feel free to just ignore :)